tests/playbooks/tests_route_next_hop_ethernet.yml

```diff
@@ -0,0 +1,191 @@
+                autoconnect: true
+                ip:
+                  dhcp4: false
+                  address:
+                    - 198.51.100.6/24
+                    - 2001:db8::4/32
+                  route:
+                    - network: 198.51.12.128
+                      prefix: 26
+                      gateway: 198.51.100.1
+                      metric: 2
+        - name: Get the IPv4 routes from the route table main
+          command: ip -4 route
+          register: route_table_main_ipv4
+          ignore_errors: true
+          changed_when: false
+
+        - name: Get the IPv6 routes from the route table main
+          command: ip -6 route
+          register: route_table_main_ipv6
+          ignore_errors: true
+          changed_when: false
+
+        - name: Assert that the route table main contains the specified IPv4
+            routes
+          assert:
+            that:
+              # When running with nm provider, the route will be configured
+              # with `proto static`
+              # In RHEL-6.10 managed host, the attribute in the route is not
+              # equally spaced
+              - route_table_main_ipv4.stdout is search("198.51.10.64/26 via
+                198.51.100.6 dev ethtest0\s+(proto static )?metric 4")
+              - route_table_main_ipv4.stdout is search("198.51.12.128/26 via
+                198.51.100.1 dev ethtest1\s+(proto static )?metric 2")
+            msg: "the route table main does not contain the specified IPv4
+              route"
+
+        - name: Assert that the route table main contains the specified IPv6
+            routes
+          assert:
+            that:
+              - route_table_main_ipv6.stdout is search("2001:db6::4 via
+                2001:db8::1 dev ethtest0\s+(proto static )?metric 2")
+            msg: "the route table main does not contain the specified IPv6
+              route"
+        - name: Get the interface1 MAC address
+          command: cat /sys/class/net/{{ interface1 }}/address
+          register: interface1_mac
+          ignore_errors: true  # noqa ignore-errors
+          changed_when: false
+          when: network_provider == "initscripts"
+        - name: Configure the network with only the MAC address specified
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface1 }}"
+                mac: "{{ interface1_mac.stdout }}"
+                state: up
+                type: ethernet
+                autoconnect: true
+                ip:
+                  dhcp4: false
+                  address:
+                    - 198.51.100.4/24
+                    - 2001:db8::2/32
+                  route:
+                    - network: 198.58.10.64
+                      prefix: 26
+                      gateway: 198.51.100.102
+                      metric: 4
+          ignore_errors: true  # noqa ignore-errors
```

Comment:
It seems like that the udev rule needs to be changed in order to bring up the connection with only the MAC specified.  I will keep investigate later.

```
# ifup /etc/sysconfig/network-scripts/ifcfg-ethtest1
WARN      : [ifup] You are using 'ifup' script provided by 'network-scripts', which are now deprecated.
WARN      : [ifup] 'network-scripts' will be removed in one of the next major releases of RHEL.
WARN      : [ifup] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well.
ERROR     : [/etc/sysconfig/network-scripts/ifup-eth] Device ethtest1@peerethtest1 does not seem to be present, delaying initialization.
```

Reply:
While this test could be a nice to have I think it is not needed. Checking that the device is being specified in the route when `interface_name` is set for `initscripts` provider should be enough if this is a problem. @tyll what do you think?

Reply:
IIRC there are two acceptance criteria (it works with interface name, there is a warning if it is missing), they should both be tested. FYI: The most modern test format for the role is in https://github.com/linux-system-roles/network/blob/main/tests/playbooks/tests_states.yml - unfortunately, it is using "user stories" instead of "acceptance criteria" to describe the tests because I learned about acceptance criteria later.

lsr_setup corresponds to the given part, lsr_test is the when part and lsr_assert_* the then part of an acceptance criterion.

Reply:
Also, I am not 100% sure what you are asking, I don't have the time to read the full test. Adjusting udev seems to be wrong, if veths are a problem, maybe try dummy interfaces.

Reply:
I think udev shouldn't be adjusted. I think use dummy interfaces is a great idea. Let's do that. Thank you!

Reply:
In the current role, the dummy interface is not supported by the initscripts provider, and the mac address is only allowed for infiniband or ethernet

Reply:
Since we are testing the warning message, we can just omit the `state: up`, and the warning message will also be logged.